wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -220,156 +173,115 @@ class PresentationActivity : FragmentActivity() {
+                    deviceRetrievalHelper = DeviceRetrievalHelper
+                        .Builder(
+                            applicationContext,
+                            deviceRetrievalHelperListener,
+                            ContextCompat.getMainExecutor(applicationContext),
+                            eDeviceKey!!
+                        )
+                        .useForwardEngagement(transport!!, deviceEngagement!!, handover!!)
+                        .build()
 
-                override fun onDeviceRequest(deviceRequestBytes: ByteArray) {
-                    Logger.i(TAG, "onDeviceRequest")
-                    deviceRequest = deviceRequestBytes
-                    state.value = State.REQUEST_AVAILABLE
                 }
 
-                override fun onDeviceDisconnected(transportSpecificTermination: Boolean) {
-                    Logger.i(TAG, "onDeviceDisconnected $transportSpecificTermination")
-                    deviceRetrievalHelper?.disconnect()
-                    deviceRetrievalHelper = null
-                    state.value = State.NOT_CONNECTED
+                State.REQUEST_AVAILABLE -> {
+                    Logger.i(TAG, "State: Request Available")
+
+                    /**
+                     * Device request bytes have been transmitted via device retrieval helper listener,
+                     * parse the bytes and generate a [PresentationRequestData] that is used throughout
+                     * all the prompts.
+                     *
+                     * Return this generated data through the callback [onPresentationRequestData],
+                     * or an Exception is thrown.
+                     */
+                    try {
+                        val presentationRequestData = parseDeviceRequestBytes(getDeviceRequest())
+                        onPresentationRequestData(presentationRequestData)
+                    } catch (exception: Exception) {
+                        Logger.e(TAG, "Unable to parse device Request Bytes: $exception")
+                    }
                 }
 
-                override fun onError(error: Throwable) {
-                    Logger.i(TAG, "onError", error)
-                    deviceRetrievalHelper?.disconnect()
-                    deviceRetrievalHelper = null
-                    state.value = State.NOT_CONNECTED
+                State.RESPONSE_SENT -> {
+                    Logger.i(TAG, "State: Response Sent")
+                    // cleanup
+                    disconnect()
                 }
 
-            },
-            ContextCompat.getMainExecutor(applicationContext),
-            eDeviceKey!!
-        )
-            .useForwardEngagement(transport!!, deviceEngagement!!, handover!!)
-            .build().apply {
-                // set deviceRetrievalHelper to this newly built instance
-                deviceRetrievalHelper = this
-                // return the instance
+                else -> {}
```

Comment:
Is this actually necessary? I thought all the possible states are handled here...